adm/src/commands/keygen.rs

```diff
@@ -18,7 +18,10 @@
 use std::ffi::CString;
 use std::fs::{metadata, OpenOptions};
 use std::io::prelude::*;
+#[cfg(target_os="linux")]
 use std::os::linux::fs::MetadataExt;
+#[cfg(target_os="macos")]
```

Comment:
Maybe this should be "not Linux", so there is at least a fallback (might be relevant for FreeBSD, for example).